src/abstractdatatable/abstractdatatable.jl

```diff
@@ -227,7 +227,7 @@ Base.ndims(::AbstractDataTable) = 2
 ##############################################################################
 
 Base.similar(dt::AbstractDataTable, dims::Int) =
-    DataTable(Any[similar(x, dims) for x in columns(dt)], copy(index(dt)))
+    DataTable(Any[similar_nullable(x, dims) for x in columns(dt)], copy(index(dt)))
```

Comment:
So the issue is that if you're calling `similar` recursively on each column, for straight up `Vector{T}` or `Vector{Union{T, Null}}`, the elements end up as `#undef`, which currently doesn't play nice with e.g. `show`. I think this is a sensible default where `similar` has a good default behavior regardless of underlying column types.